src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Segment.java

```diff
@@ -2,12 +2,31 @@
 
 import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
 public abstract class Segment {
     protected static final String PLUS_SEPARATOR = "+";
     protected final static String COLON_SEPARATOR = ":";
     protected final static String COMMA_SEPARATOR = ",";
     private static final String TERMINATOR = "'";
 
+    protected static <T> List<T> removeEmptyLeadingFields(List<T> list, Predicate<T> predicate) {
```

Comment:
Should this be named removeEmpty**Trailing**Fields - it removes things from the end right?